src/Query.tsx

```diff
@@ -443,8 +407,11 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
       const { fetchPolicy } = this.queryObservable!.options;
       Object.assign(data, { loading, networkStatus, error });
 
+      const previousData =
+        this.lastRenderedResult ? this.lastRenderedResult.data : {};
+
       if (loading) {
-        Object.assign(data.data, this.previousData, currentResult.data);
+        Object.assign(data.data, previousData, currentResult.data);
```

Comment:
It's confusing that `data` here has a `data.data` property. Can we call the `data` variable something like `result` instead? This method is called `getQueryResult`, after all.